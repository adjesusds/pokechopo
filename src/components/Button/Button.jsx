// External Packages
import { bool, func, node } from 'prop-types'
import React from 'react'

// Internal Packages
import StyledButton from './styled/Button'

/**
 * Creates a new Button component.
 * @class
 */
const Button = ({ children, onClick, disabled }) => (
  <StyledButton onClick={onClick} disabled={disabled}>
    {children}
  </StyledButton>
)

/** @override */
Button.propTypes = {
  children: node.isRequired,
  onClick: func.isRequired,
  disabled: bool,
}

/** @override */
Button.defaultProps = {
  disabled: false,
}

export default Button
