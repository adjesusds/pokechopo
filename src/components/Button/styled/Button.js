// External Packages
import styled, { css } from 'styled-components'

// Internal Packages
import { colors, dimensions, durations } from 'config/palette'

/**
 * @constant {Element} style The style of the component.
 */
const style = styled.button`
  background-color: ${colors.black};
  border: 2px solid ${colors.white};
  border-radius: 3px;
  color: ${colors.white};
  cursor: pointer;
  padding: ${dimensions.xMedium};
  transition: all ${durations.fast} ease-in;
  transition-property: border-color, color;
  ${({ disabled }) =>
    disabled &&
    css`
      border-color: ${colors.grey.x500};
      color: ${colors.grey.x500};
      cursor: wait;
    `};

  &:focus {
    outline: none;
  }
`

export default style
