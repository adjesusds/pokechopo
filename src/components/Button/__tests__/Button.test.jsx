// External Packages
import { shallow } from 'enzyme'
import React from 'react'

// Internal Packages
import Button from '../Button'

describe('<Button />', () => {
  it('should render the component on valid props', () => {
    let component

    component = shallow(<Button onClick={() => {}}>Hello, World!</Button>)

    expect(component).toMatchSnapshot()

    component = shallow(
      <Button onClick={() => {}} disabled>
        Hello, World!
      </Button>
    )

    expect(component).toMatchSnapshot()
  })
})
