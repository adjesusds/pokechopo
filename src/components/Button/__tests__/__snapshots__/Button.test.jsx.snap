// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the component on valid props 1`] = `
<styled.button
  disabled={false}
  onClick={[Function]}
>
  Hello, World!
</styled.button>
`;

exports[`<Button /> should render the component on valid props 2`] = `
<styled.button
  disabled={true}
  onClick={[Function]}
>
  Hello, World!
</styled.button>
`;
